sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/csv/CSVInferSchema.scala

```diff
@@ -86,6 +86,7 @@ private[csv] object CSVInferSchema {
         case IntegerType => tryParseInteger(field)
         case LongType => tryParseLong(field)
         case DoubleType => tryParseDouble(field)
+        case _: DecimalType => tryParseDecimal(field)
```

Comment:
I added the `_` because `DecimalType` looks referencing the companion object. I tried that expression before but this emits the compilation error below.

```
Error:(89, 14) pattern type is incompatible with expected type;
 found   : org.apache.spark.sql.types.DecimalType.type
 required: org.apache.spark.sql.types.DataType
Note: if you intended to match against the class, try `case DecimalType(_,_)`
        case DecimalType => tryParseDecimal(field)
             ^
```